shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java

```diff
@@ -192,4 +207,16 @@ private void processingComplete(Class<?> entity) {
     private boolean isSuccessfulResponse(HttpResponse<?> httpResponse) {
         return httpResponse.statusCode() >= 200 && httpResponse.statusCode() < 400;
     }
+
+    private void updateManagerRequestMetricsOnSuccess(ManagerRequestType requestType, HttpResponse<Buffer> successResponse) {
+        metricsService.updateManagerRequestMetrics(requestType, ManagerRequestStatus.SUCCESS, String.valueOf(successResponse.statusCode()));
+    }
+
+    private void updateManagerRequestMetricsOnFailure(ManagerRequestType requestType, Throwable error) {
+        if (error instanceof HTTPResponseException) {
+            int statusCode = ((HTTPResponseException) error).getStatusCode();
+            metricsService.updateManagerRequestMetrics(requestType, ManagerRequestStatus.FAILURE, String.valueOf(statusCode));
+        }
+        metricsService.updateManagerRequestMetrics(requestType, ManagerRequestStatus.FAILURE, StringUtils.EMPTY);
+    }
```

Comment:
I think we should keep track of the failures also when they are not instanceof `HTTPResponseException`

Reply:
I think we do, we're just not extracting any HTTP Response code from them.

Reply:
I think we do, we're just not extracting any HTTP Response code from them.